cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/BerryClient.scala

```diff
@@ -63,12 +63,14 @@ class BerryClient(val jsonParser: JSONParser,
           // TODO send error messages to user
           throw JsonRequestException("Batch Requests cannot contain \"limit\" field")
         }
-        //Each streaming request need a specific actor to handle the request.
-        //TODO Ultimately, clients can run multiple streaming request simultaneously.
-        //     They can also cancel or reset a specific request.
-        //     Right now, we are just allow one streaming request at once, the later one will stop the previous running request.
-        val child = context.child("stream").getOrElse(
-          context.actorOf(Props(new ProgressiveSolver(dataManager, planner, config, out)), "stream")
+        //Right now, we create one stream actor for one 'category' query indicated by 'transform'->'wrap'->'category'.
+        //TODO Clients can also cancel or reset a specific request.
+        var streamActorName = "default"
```

Comment:
To avoid using `var`, you can write the code like following: 
```
val actorName = transform match {
   case categorical : ICategoricalTransform => categorical. category
   case _ : "default"
}
```

Reply:
Great! That's what I am looking for.